server/bot/commands/trivia.ts

```diff
@@ -0,0 +1,278 @@
+import { Message, GuildTextableChannel, TextableChannel, User, EmbedOptions, Client } from 'eris'
+import { Command, DB } from '../imports/types'
+import fs from 'fs'
+import { promisify } from 'util'
+import { getInsult } from '../imports/tools'
+
+async function parseTriviaList (fileName: string) {
+  const readFile = promisify(fs.readFile)
+  const data = await readFile(`./triviaLists/${fileName}.txt`, 'utf8')
+  const triviaList = data.split('\n')
+  return triviaList
+    .map(el => el.replace('\n', '').split('`'))
+    .filter(el => el.length >= 2 && el[0])
+    .map(el => ({
+      question: el[0],
+      answers: el.slice(1).map(ans => ans.trim())
+    }))
+}
+
+export class TriviaSession {
+  revealMessages = ['I know this: ', 'Easy: ', 'Of course, it\'s: ']
+  failMessages = ['You suck at this', 'That was trivial, really', 'Moving on...']
+  settings: { maxScore: number, timeout: number, delay: number, botPlays: boolean, revealAnswer: boolean }
+  currentLine: {question: string, answers: string[]}
+  channel: TextableChannel
+  author: User
+  message: Message
+  questionList: {question: string, answers: string[]}[]
+  scores: { [char: string]: number } = {}
+  status = ''
+  timer: number
+  timeout: number
+  count: number
+  tempDB: DB
+  client: Client
+
+  constructor (triviaList: {question: string, answers: string[]}[], message: Message, botPlays: boolean, timeLimit: number, maxScore: number, revealAnswers: boolean, tempDB: DB, client: Client) {
```

Comment:
We can reduce this constructor in size quite a bit?
- `this.count`, `this.timer` and `this.timeout` could be moved up into the field definitions.
- `triviaList` can become an interface to reduce line length.
- `this.settings = { maxScore, timeout: 120000, timeLimit, botPlays, revealAnswers }` (rename `revealAnswer` and `delay` in the object to `revealAnswers` and `timeLimit` for this to work).

Reply:
If switching to Map, just do this then, no need for interface.
```suggestion
  constructor (triviaList: Map<string, string[]>, message: Message, botPlays: boolean, timeLimit: number, maxScore: number, revealAnswer: boolean, tempDB: DB, client: Client) {
```
Also, `revealAnswer` or `revealAnswers`? lol consider updating this locally, I won't be spamming patches here to rename that, it's tiny and just something I noticed.